backend/app/core/validators/utils/language_detector.py

```
@@ -0,0 +1,49 @@
+        self.lid = pipeline(
+            task = "text-classification",
+            model="papluca/xlm-roberta-base-language-detection",
+            top_k=1
+        )
+        self.label = None
+
+    @staticmethod
+    def _normalize(label: str) -> str:
+        return LANG_HINDI if label.startswith(LANG_HINDI) else label
+
+    @lru_cache(maxsize=1024)
```

Ruff (0.14.8)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)